native/src/hooks/useUserLocation.ts

```diff
@@ -35,6 +32,22 @@ const getUserLocation = async (): Promise<LocationStateType> =>
     )
   })
 
+const locationPermissionIOS = PERMISSIONS.IOS.LOCATION_WHEN_IN_USE
+const fineLocationPermissionAndroid = PERMISSIONS.ANDROID.ACCESS_FINE_LOCATION
+const coarseLocationPermissionAndroid = PERMISSIONS.ANDROID.ACCESS_COARSE_LOCATION
+
+const getLocationPermissionStatus = async (requestPermission: boolean) => {
+  const permissionFunctionToCall = requestPermission ? request : check
```

Comment:
:upside_down_face: I think this naming is quite confusing, but might be a personal opinion
```suggestion
  const checkOrRequest = requestPermission ? request : check
```